api-server/src/server/boot/donate.js

```diff
@@ -42,117 +46,84 @@ export default function donateBoot(app, done) {
     });
   }
 
-  function createStripeDonation(req, res) {
-    const { user, body } = req;
+  async function createStripeDonation(req, res) {
+    const { body } = req;
+    const { amount, duration, email, subscriptionId } = body;
+    try {
+      const subscription = await stripe.subscriptions.retrieve(subscriptionId);
+      const isSubscriptionActive = subscription.status === 'active';
+      const productId = subscription.items.data[0].plan.product;
+      const isSubscribedInMinutes = isWithinFiveMinutes(
```

Comment:
Why "within"? Surely we want to know that `current_period_start` was in the _last_ five minutes, no?

Reply:
The main purpose for this to check if it has happened recently. 5 minutes is only an arbitrary number. 
Do you suggest calling it inLastFiveMinutes?

Reply:
That sounds reasonable. To be honest, it was the fact that `isWithinFiveMinutes` checked the absolute time difference that got me wondering. That seemed very weird!